api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -453,6 +453,40 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  message LocalReplyConfig {
+    // Configuration of list of mappers which allows to filter and change local response. 
+    // Code iterate through mappers until first match of filter.
+    repeated ResponseMapper mapper = 1;
+
+    // Allows to define custom format of local reply.
+    oneof reply_format {
+      // Response body plain text format.
+      string format = 2;
+  
+      // Response body json format as dictionary.
+      google.protobuf.Struct json_format = 3;
+    }
+  };
+
+  // Configuration of local reply returned by Envoy. Allows to specify mapping and format of response.
+  LocalReplyConfig local_reply_config = 35;
+}
+
+message ResponseMapper {
+  // Filter is used to determine if the response should be changed.
+  envoy.config.filter.accesslog.v2.AccessLogFilter filter = 1 [(validate.rules).message.required = true];
```

Comment:
Great, if this works I think adding #next-major-version comment in `AccessLogFilter` to rename it more generically.

Reply:
Yep, this sounds great.